docs/api/exceptions.rst

```diff
@@ -0,0 +1,39 @@
+
+.. xdata:: datatable.exceptions
+    :src: --
+
+    This namespace contains warnings and exceptions that ``datatable`` may
+    throw during runtime.
+
+    Warnings are issued when it is helpful to inform the user of some condition
+    in a program, that doesn't result in an exception and the program termination.
+    Along with the built-in Python `warnings <https://docs.python.org/3/library/warnings.html>`_
+    ``datatable`` may also issue:
+
+    .. list-table::
+        :widths: auto
+        :class: api-table
+
+        * - ``DatatableWarning``
+          - Generic ``datatable`` warnings;
+
+        * - ``FreadWarning``
+          - Warnings related to :func:`fread`;
+
+        * - `IOWarning`
+          - Warnings related to input/output.
+
+
+    Exceptions are thrown when a special condition, that is unexpected or
+    anomalous, encountered during runtime. Along with the built-in Python
+    `exceptions <https://docs.python.org/3/library/exceptions.html>`_
+    ``datatable`` may also throw:
+
+    .. list-table::
+        :widths: auto
+        :class: api-table
+
+        * - `InvalidOperationError`
+          - Thrown when the operation requested is illegal
+            for the given combination of parameters.
```

Comment:
There are many more exceptions, see `exceptions.py`.

Reply:
Yes, but they're all the same as the built-in Python exceptions.

Reply:
They are not exactly the same: they inherit from the builtin exception with the same name, and also from the common `DtException`, which means that you can catch any `datatable` error by catching `DtException`.

Plus, there's still advantage in listing all exceptions that datatable can throw, and describing general circumstances when different exceptions may occur. For example, you mentioned that we should document the fact that argument types for each function are strictly enforced. The perfect place for that is the docpage for our `TypeError`.

Reply:
>which means that you can catch any datatable error by catching DtException

This is definitely worth to mention. But listing all the exceptions with the same names as the built-in ones seems like just reiteration of the standard python docs. For instance, `TypeError` has exactly the same meaning in `datatable` and in python as explained here https://docs.python.org/3/library/exceptions.html#TypeError